4/m_ipinfo_io.cpp

```diff
@@ -0,0 +1,189 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *   Copyright (C) 2024 Jean Chevronnet <mike.chevronnet@gmail.com>
+ *
+* This file contains a third party module for InspIRCd.  You can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/// $ModAuthor: Jean Chevronnet (reverse) <mike.chevronnet@gmail.com>
+/// $ModDesc: Ip information from Ipinfo.io in /WHOIS (only irc operators), found more information at https://ipinfo.io/developers.
+/// $ModDepends: core 4
+/// $ModConfig: <ipinfo apikey="YOUR IP INFO.IO APIKEY">
+/// $CompilerFlags: find_compiler_flags("RapidJSON")
+/// $CompilerFlags: find_compiler_flags("libcurl")
+/// $LinkerFlags: find_linker_flags("libcurl")
+
+#include "inspircd.h"
+#include "extension.h"
+#include "modules/httpd.h"
+#include "modules/stats.h"
+#include "modules/whois.h"
+#include <rapidjson/document.h>
+#include <rapidjson/error/en.h>
+#include <curl/curl.h>
+#include <thread>
+#include <mutex>
+
+enum
+{
+    // deff custom numeric for this information in whois
+    RPL_WHOISIPINFO = 695,
```

Comment:
You may find it better to use `RPL_WHOISSPECIAL` (320) here as it is more of a generic numeric for text strings relating to a user that doesn't need special client support.